cards.js

```diff
@@ -1,23 +1,69 @@
-var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
-var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
-
-function shuffle(array){
-  var currentIndex = array.length;
-  var randomIndex;
-  var temporaryValue;
-  while(currentIndex > 0){
-    randomIndex   = Math.floor(Math.random() * currentIndex);
-    currentIndex -= 1;
-
-    temporaryValue       = array[currentIndex];
-    array[currentIndex]  = array[randomIndex];
-    array[randomIndex]   = temporaryValue;
+  var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"]; // all the values in the deck
```

Comment:
Be mindful of your indentation! These first ~20 lines are indented 1 level, but they shouldn't be.